tests/prague/eip7692_eof_v1/eip3540_eof_v1/test_all_opcodes_in_container.py

```diff
@@ -129,3 +131,271 @@ def test_all_opcodes_in_container(
+
+    eof_test(
+        data=Container(
+            sections=sections,
+        ),
+        expect_exception=EOFException.UNREACHABLE_INSTRUCTIONS
+        if opcode != Op.RETURNCONTRACT
+        else EOFException.INCOMPATIBLE_CONTAINER_KIND,
+    )
+
+
+@pytest.mark.parametrize(
+    "opcode",
+    [op for op in list(Op) if op not in invalid_eof_opcodes and op.min_stack_height > 0]
+    + [
+        # Opcodes that have variable min_stack_height
+        Op.SWAPN[0x00],
+        Op.SWAPN[0xFF],
+        Op.DUPN[0x00],
+        Op.DUPN[0xFF],
+        Op.EXCHANGE[0x00],
+        Op.EXCHANGE[0xFF],
+    ],
+)
+def test_all_opcodes_stack_underflow(
+    eof_test: EOFTestFiller,
+    opcode: Opcode,
+):
+    """
+    Test stack underflow on all opcodes that require at least one item on the stack
+    """
+    sections: List[Section]
+    if opcode == Op.EOFCREATE:
+        sections = [
+            Section.Code(code=Op.ORIGIN * (opcode.min_stack_height - 1) + opcode[0] + Op.STOP),
+            Section.Container(
+                container=Container(
+                    sections=[
+                        Section.Code(code=Op.RETURNCONTRACT[0](0, 0)),
+                        Section.Container(Container.Code(code=Op.STOP)),
+                    ]
+                )
+            ),
+        ]
+    elif opcode == Op.RETURNCONTRACT:
+        sections = [
+            Section.Code(code=Op.EOFCREATE[0](0, 0, 0, 0) + Op.STOP),
+            Section.Container(
+                container=Container(
+                    sections=[
+                        Section.Code(code=Op.ORIGIN * (opcode.min_stack_height - 1) + opcode[0]),
+                        Section.Container(Container.Code(code=Op.STOP)),
+                    ]
+                )
+            ),
+        ]
+    else:
+        bytecode = Op.ORIGIN * (opcode.min_stack_height - 1)
+        if opcode.has_data_portion():
+            bytecode += opcode[0]
+        else:
+            bytecode += opcode
+        bytecode += Op.STOP
+        sections = [Section.Code(code=bytecode)]
+    eof_code = Container(sections=sections)
+
+    eof_test(
+        data=eof_code,
+        expect_exception=EOFException.STACK_UNDERFLOW,
+    )
+
+
+@pytest.mark.parametrize(
+    "opcode",
+    [
+        op
+        for op in list(Op)
+        if op not in invalid_eof_opcodes and op.pushed_stack_items > op.popped_stack_items
+    ]
+    + [
+        Op.DUPN[0x00],
+        Op.DUPN[0xFF],
+    ],
+)
+@pytest.mark.parametrize(
+    "exception",
+    [EOFException.INVALID_MAX_STACK_HEIGHT, EOFException.MAX_STACK_HEIGHT_ABOVE_LIMIT],
+)
+def test_all_opcodes_stack_overflow(
+    eof_test: EOFTestFiller,
+    opcode: Opcode,
+    exception: EOFException,
+):
+    """
+    Test stack overflow on all opcodes that push more items than they pop
+    """
+    opcode_count = ceiling_division(
+        MAX_OPERAND_STACK_HEIGHT + 1 - opcode.min_stack_height,
+        opcode.pushed_stack_items - opcode.popped_stack_items,
```

Comment:
This may be over-generalized. So far "pushing" instruction increase the stack height only by 1. Maybe use assert instead for future proofing?

Reply:
`opcode.pushed_stack_items` is guaranteed to be higher than `opcode.popped_stack_items` by the `pytest.mark.parametrize` condition, or do you mean some other assertion?

Reply:
```suggestion
        assert opcode.pushed_stack_items - opcode.popped_stack_items == 1
        opcode_count = MAX_OPERAND_STACK_HEIGHT + 1 - opcode.min_stack_height
```

Reply:
Applied on latest version, thanks.